### PR TITLE
Document allow outdated vendors mode

### PR DESCRIPTION
**Pull request by greg0ire**
Refs https://github.com/symfony/symfony/pull/24867

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Comment by greg0ire** [comment]
I wonder if should add a table to recommend a usage:

- `strict`: recommended for actively maintained projects with few dependencies
- `<some_integer>` recommended to transition towards strictness
- `weak_lagging_vendors`: recommended in projects when strict cannot be attained
- `weak_vendors`: recommended in libraries that struggle to keep up with deprecations
- `weak`: not recommended

**Review by rzilin — APPROVED** [review]

**Comment by greg0ire** [comment]
Here is how the table looks:

screenshot-2018-1-29 the phpunit bridge symfony framework documentation documentation

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -188,6 +188,39 @@ the ``vendors`` directory will make the test suite fail, while deprecations
...
 worlds.
 
+When working on a project, you might be more interested in the
+``allow_outdated_vendors``, which is a bit more strict. Let's say you
+want to fix deprecations as soon as they appear. A problem many people
+experience is that some dependencies they have tend to lag behind their
+own dependencies, meaning they do not fix deprecations as soon as
+possible, which means there is nothing you can do to fix that (apart
+from a pull request on the outdated vendor). This mode allows you to
+ignore those deprecations, allowing you to notice when your code is
```
Comment (OskarStark) [inline components/phpunit_bridge.rst]: maybe display `your code` bold?

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -188,6 +188,39 @@ the ``vendors`` directory will make the test suite fail, while deprecations
...
+using deprecated APIs, and to keep up with the changes.
+
+Here is a summary that should help you pick the right mode:
+
++------------------------+-----------------------------------------------------+
+| Mode                   | Recommended situation                               |
++========================+=====================================================+
```
Comment (OskarStark) [inline components/phpunit_bridge.rst]: `Recommended ` repeats a lot

File: components/phpunit_bridge.rst (hunk 1 above)
Comment (greg0ire) [reply components/phpunit_bridge.rst]: Or even better, *italics*.

File: components/phpunit_bridge.rst (hunk 2 above)
Comment (greg0ire) [reply components/phpunit_bridge.rst]: Yeah, not sure what to do about that…

**Comment by javiereguiluz** [comment]
@greg0ire the related PR (https://github.com/symfony/symfony/pull/24867) was closed. Could you please tell us if this PR is still relevant or another one is needed to document the changes mentioned in the closed PR? Thanks!

**Comment by greg0ire** [comment]
@javiereguiluz this should be closed for now, sorry for leaving it open.

**Pull request closed without merging**